cmd/utils/cmd.go

```diff
@@ -312,3 +313,28 @@ func ExportPreimages(db ethdb.Database, fn string) error {
 	log.Info("Exported preimages", "file", fn)
 	return nil
 }
+
+func ensureSufficientMemory(sigc chan os.Signal) {
+	for t := range time.Tick(5 * time.Second) {
+		select {
+		case <-sigc:
+			log.Info("Node is being terminated...")
+		default:
+			go func() {
+				var stat syscall.Statfs_t
+				wd, err := os.Getwd(); err != nil {
+					Fatalf("Error reading available memory of Node: %v", err)
+				}
+				syscall.Statfs(wd, &stat)
+				MB = 1000000
+				avMemMB = stat.Bavail * uint64(stat.Bsize) / MB
+				if avMemMB < 100 {
+					log.Info("Available disk space is less than 100 MB. Gracefully shutting down to prevent database corruption.")
+					sigc <- syscall.SIGTERM
+				} else if avMemMB < 500 {
+					log.Warnf("Node is running low on memory. It will terminate if memory runs below 100MB. Remaining: %v MB.", avMemMB)
```

Comment:
Same here, `low on disk space ... if disk space runs below... `

Reply:
Addressed in c355ae4c3b6bd709e781d37051814f3586130d9f